### ISSUE TITLE
[Linux] melonDS.ini created with \ in name if no existing config is found

### ISSUE THREAD

**Issue by Ace4896** [Arisotura/melonDS#413]
If I run the app via a terminal (like `./melonDS`) and change some settings, the config file is given the wrong name:

```bash
jon@jon-acer-aspire-a715-71g:~/Documents/Portable Apps/MelonDS> ./melonDS
melonDS 0.7.4
http://melonds.kuribo64.net/
jon@jon-acer-aspire-a715-71g:~/Documents/Portable Apps/MelonDS> ls -a1
.
..
bios7.bin
bios9.bin
firmware.bin
firmware.bin.bak
melonDS
.\melonDS.ini
romlist.bin
timewarp.mln
```

For context, I ran into this issue while I was attempting to bundle melonDS as an AppImage. It created the config file in the parent directory rather than the app's directory, so I decided to check what would happen normally. I'm still figuring this out though, so I'll make a PR if I manage to get it working properly.

**Comment by Arisotura** [comment]
weird. can't reproduce. Ubuntu 18.04

**Comment by Ace4896** [comment]
iirc the config file is checked for in several locations right? I think its because it couldn't find the config file in these locations and had to create a new one, but the wrong path separator was chosen.

I encountered this a while back on my KDE Neon system by accident, but I didn't know how to reproduce it. I can reliably reproduce it in my openSUSE Tumbleweed system now, so I'll try to narrow this down further.

**Comment by Ace4896** [comment]
OK, I did a bit of debugging and I think I've figured out what's happening. I've been doing this on commit 99b07f930084b6faa7ac130b0d2c82a6f2060306.

In [Platform.cpp, line 171](https://github.com/Arisotura/melonDS/blob/99b07f930084b6faa7ac130b0d2c82a6f2060306/src/libui_sdl/Platform.cpp#L171), `emudirpath` has a single backslash for the final path separator, e.g. `/home/jon/Documents/Repositories/GitHub/MelonDS/Ace4896/build\\melonDS.ini`. I'm guessing this backslash works as intended on Windows since it uses that backslash for the path separator, but on Linux it becomes part of the file name instead.

Changing that line to:

```c
emudirpath[emudirlen] = '/';
```

fixed it, so maybe there needs to be an `#ifdef __WIN32__` thing to handle the different path separators.
